les/vflux/client/fillset_test.go

```diff
@@ -43,7 +43,26 @@ func (i *testIter) Node() *enode.Node {
 	return i.node
 }
 
-func (i *testIter) Close() {}
+func (i *testIter) Close() {
+loop1:
+	for {
+		select {
+		case <-i.nodeCh:
+		default:
+			break loop1
+		}
+	}
+	close(i.nodeCh)
+loop2:
+	for {
+		select {
+		case <-i.waitCh:
+		default:
+			break loop2
+		}
+	}
+	close(i.waitCh)
```

Comment:
I'm not sure `close` is needed - may cause pancs due to send on closed channel. And also, I think this is semantically equivalent (minus closing): 

```suggestion
	for {
		select {
		case <-i.nodeCh:
		case <-i.waitCh:
		default:
			return
		}
	}
```
(and if we do want to close, then we can do the two `close` just before the return)

Reply:
Wow, your suggestion is cleaner and better.  :+1: 
I think we need to call `close()` before the return, because we donn't need `testIter` any more after we call `testIter`.